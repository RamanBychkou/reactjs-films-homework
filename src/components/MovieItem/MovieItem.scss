.movieContainer {
  height: 31.9rem;
  position: relative;
  margin: 0.5rem;
  border-radius: 5px;
  overflow: hidden;

  img {
    object-fit: cover;
    width: 100%;
  }

  figure {
    height: 100%;
  }

  .movieGenres ul {
    display: flex;
    justify-content: flex-start;
    flex-flow: row wrap;

    button {
      background: none;
      border: none;
      padding: 0;
      color: #47c3f0;
      cursor: pointer;
    }

    li:hover,
    li:focus {
      color: #000;
    }
  }

  figcaption {
    background: #fff;
    height: 20%;
    display: flex;
    flex-flow: column nowrap;
    justify-content: baseline;
    align-items: inherit;
    padding: 1rem 1rem 0 1rem;

    .movieTitle {
      display: flex;
      justify-content: space-between;
      align-items: baseline;
      white-space: nowrap;
      text-overflow: ellipsis;
      margin-bottom: 0.5rem;

      button,
      a {
        text-transform: uppercase;
        text-decoration: none;
        color: #000;
        font-weight: bold;
        white-space: nowrap;
        text-overflow: ellipsis;
        overflow: hidden;
        margin-right: 0.3rem;
        background: none;
        border: none;
        padding: 0;
        outline: none;

        &:hover,
        &:focus {
          color: #47c3f0;
          white-space: pre-wrap;
        }
      }
    }
  }

  &:hover {
    figure img {
      height: 100%;
    }
  }
}

figure > div {
  height: 80%;
  overflow: hidden;
}

.cards {
  width: 300px;
}

.lines {
  width: 100%;

  & > div {
    width: 100%;
  }
}

.figcaptionNone figcaption {
  display: none !important;
}

.figcaptionNone > div {
  height: 100%;
}

.figcaptionBlock figcaption {
  display: block;
}

.figcaptionNone + div {
  height: 100%;
}
