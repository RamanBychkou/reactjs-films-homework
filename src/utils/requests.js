import itemsIsLoadingAction from '../modules/isLoading/isLoadingAction';
import { itemsFetchDataSuccess, fetchVideoSuccess, clearResults } from '../modules/root/rootAction';
import setMainMovieDetails from '../modules/mainMovie/mainMovieAction';
import itemsHasErrored from '../modules/Error/errorAction';

const KEY = '75331f1a740385460b25b56203149aa8';

const requestsFilms = {
  fetchListMovies: (dispatch, query) => {
    let url;
    // const id = query.replace(/\/[genre]/g);
    switch (query) {
      case 'coming_soon':
        url = `https://api.themoviedb.org/3/movie/upcoming?api_key=${KEY}&language=en-US&page=1`;
        break;
      case 'trending':
        url = `https://api.themoviedb.org/3/movie/popular?api_key=${KEY}&language=en-US&page=1`;
        break;
      case '/':
        url = `https://api.themoviedb.org/3/movie/popular?api_key=${KEY}&language=en-US&page=1`;
        break;
      case 'top_rated':
        url = `https://api.themoviedb.org/3/movie/top_rated?api_key=${KEY}&language=en-US&page=1`;
        break;
      default:
        url = `https://api.themoviedb.org/3/discover/movie?api_key=${KEY}&with_genres=${query}`;
    }
    dispatch(itemsIsLoadingAction(true));
    return fetch(url)
      .then(resp => resp.json())
      .then((data) => {
        dispatch(itemsIsLoadingAction(false));
        if (data.ok === false) {
          throw new Error('Something wrong');
        }
        return dispatch(
          itemsFetchDataSuccess({
            page: data.page,
            results: data.results,
          }),
        );
      })
      .catch(err => dispatch(itemsHasErrored(err)));
  },
  getMainMovieDetails: (dispatch, id) => {
    dispatch(itemsIsLoadingAction(true));
    return fetch(
      `https://api.themoviedb.org/3/movie/${id}?api_key=75331f1a740385460b25b56203149aa8&language=en-US`,
    )
      .then(resp => resp.json())
      .then((data) => {
        dispatch(itemsIsLoadingAction(false));
        if (data.ok === false) {
          throw new Error('Something wrong');
        }
        return dispatch(setMainMovieDetails(data));
      })
      .catch(err => dispatch(itemsHasErrored(err)));
  },

  fetchVideo: (dispatch, id) => {
    dispatch(itemsIsLoadingAction(true));
    return fetch(`https://api.themoviedb.org/3/movie/${id}/videos?api_key=${KEY}&language=en-US`)
      .then(resp => resp.json())
      .then((data) => {
        dispatch(itemsIsLoadingAction(false));
        if (data.ok === false) {
          throw new Error('Something wrong');
        }
        let videoId = '';
        data.results.some((curr) => {
          if (curr.type === 'Trailer') {
            videoId = curr.key;
            return true;
          }
          return false;
        });
        return dispatch(fetchVideoSuccess(videoId));
      })
      .catch(err => dispatch(itemsHasErrored(err)));
  },

  fetchSearchResults: (dispatch, query) => {
    dispatch(itemsIsLoadingAction(true));
    return fetch(
      `https://api.themoviedb.org/3/search/movie?api_key=${KEY}&language=en-US&query=${query.replace(
        '%20',
        '',
      )}&page=1&include_adult=false`,
    )
      .then(resp => resp.json())
      .then((data) => {
        if (data.results.length === 0) {
          dispatch(clearResults());
          throw new Error('Nothing found');
        }
        dispatch(itemsIsLoadingAction(false));
        return dispatch(
          itemsFetchDataSuccess({
            page: data.page,
            results: data.results,
          }),
        );
      })
      .catch(err => dispatch(itemsHasErrored(err.message)));
  },
};

export default requestsFilms;
