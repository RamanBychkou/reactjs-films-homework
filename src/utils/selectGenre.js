const selectGenre = function selectGenre(props, e) {
  const {
    genres, fetchListMovies, error, clearError,
  } = props;
  const genre = e.target.value || e.target.textContent.replace(/[^A-Za-z\s]/g, '');
  let genreId;
  genres.some((a) => {
    if (genre === a.name) {
      genreId = a.id;
      return true;
    }
    return false;
  });
  if (error !== undefined && error !== '') {
    clearError(false);
  }
  return fetchListMovies(genreId);
};

export const keydonwGenres = (props, e) => {
  if (e.key === 'Enter') {
    selectGenre(props, e);
  }
  return props;
};

export default selectGenre;
