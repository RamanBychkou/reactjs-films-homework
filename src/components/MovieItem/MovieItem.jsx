import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import HoverMovieCard from '../HoverMovieCard/HoverMovieCard';
import selectGenre, { keydonwGenres } from '../../utils/selectGenre';

import * as stylerate from '../Star/Star.scss';
import style from './MovieItem.scss';

class MovieItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      typeView: 'cards',
      fullOverwie: false,
    };
  }

  static getDerivedStateFromProps(nextProps) {
    return { typeView: nextProps.typeView };
  }

  shouldComponentUpdate(nextProps, nextState) {
    const prev = this.state;
    if (JSON.stringify(prev) !== JSON.stringify(nextState)) {
      return true;
    }
    return false;
  }

  hoverOverwieBlock = () => {
    const { typeView } = this.state;
    this.setState({ typeView, fullOverwie: true });
  };

  hoverOverwieNone = () => {
    const { typeView } = this.state;
    this.setState({ typeView, fullOverwie: false });
  };

  changeMainFilm = (e) => {
    e.preventDefault();
    const { getMainMovieDetails } = this.props;
    const element = e.target;
    return getMainMovieDetails(element.id);
  };

  render() {
    const {
      curr, genres, fetchVideo, movies,
    } = this.props;
    const { typeView, fullOverwie } = this.state;
    let imageLink;
    if (curr.poster_path === null || curr.backdrop_path === null) {
      imageLink = 'https://api.ballotpedia.org/v3/thumbnail/';
    } else if (typeView === 'cards') {
      imageLink = `https://image.tmdb.org/t/p/w300${curr.poster_path}`;
    } else {
      imageLink = `https://image.tmdb.org/t/p/w1280${curr.backdrop_path}`;
    }
    const classNamefigcaption = fullOverwie === true ? style.figcaptionNone : style.figcaptionBlock;
    const imgAlt = curr.title;
    const genresRow = curr.genre_ids
      .map((c) => {
        let currentGenre;
        genres.some((a) => {
          if (c === a.id) {
            currentGenre = a.name;
            return true;
          }
          return false;
        });
        return currentGenre;
      })
      .join(', 1')
      .split(' 1')
      .map(cur => (
        <li key={`key-${cur.replace(',', '')}`}>
          <button
            type="button"
            onClick={selectGenre.bind(null, this.props)}
            onKeyDown={keydonwGenres.bind(null, this.props)}
          >
            {cur}
          </button>
        </li>
      ));

    return (
      <article
        className={`${style.movieContainer} ${style[typeView]}`}
        onMouseLeave={this.hoverOverwieNone}
      >
        <figure className={classNamefigcaption}>
          <div>
            <img src={imageLink} alt={imgAlt} />
          </div>
          <figcaption>
            <div className={style.movieTitle}>
              <button
                type="button"
                href="#"
                onClick={this.changeMainFilm}
                onKeyDown={keydonwGenres.bind(null, this.props)}
                tabIndex="0"
              >
                <Link
                  to={{
                    pathname: `/films/${curr.id}`,
                  }}
                  id={curr.id}
                >
                  {curr.title}
                </Link>
              </button>

              <span className={stylerate.rate}>{curr.vote_average}</span>
            </div>

            <div className={style.movieGenres}>
              <ul>{genresRow}</ul>
            </div>
          </figcaption>
        </figure>
        <HoverMovieCard
          overview={curr.overview}
          genres={genresRow}
          title={curr.title}
          rate={curr.vote_average}
          fetchVideo={fetchVideo}
          movies={movies}
          id={curr.id}
          fullOverwie={fullOverwie}
          hoverOverwieBlock={this.hoverOverwieBlock}
          hoverOverwieNone={this.hoverOverwieNone}
        />
      </article>
    );
  }
}

MovieItem.defaultProps = {
  fetchVideo: alert,
  movies: {},
};

MovieItem.propTypes = {
  movies: PropTypes.objectOf(PropTypes.any),
  curr: PropTypes.objectOf(PropTypes.any).isRequired,
  genres: PropTypes.arrayOf(PropTypes.object).isRequired,
  fetchVideo: PropTypes.func,
  getMainMovieDetails: PropTypes.func.isRequired,
};

export default MovieItem;
