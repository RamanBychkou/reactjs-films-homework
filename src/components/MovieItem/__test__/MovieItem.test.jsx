import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils, { act } from 'react-dom/test-utils';
import { MemoryRouter as Router } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import fetchMock from 'fetch-mock';
import MovieItem from '../index';
import { keydonwGenres } from '../../../utils/selectGenre';
import requestsFilms from '../../../utils/requests';

const genres = [
  { id: 28, name: 'Action' },
  { id: 12, name: 'Adventure' },
  { id: 16, name: 'Animation' },
  { id: 35, name: 'Comedy' },
  { id: 80, name: 'Crime' },
  { id: 99, name: 'Documentary' },
  { id: 18, name: 'Drama' },
  { id: 10751, name: 'Family' },
  { id: 14, name: 'Fantasy' },
  { id: 36, name: 'History' },
  { id: 27, name: 'Horror' },
  { id: 10402, name: 'Music' },
  { id: 9648, name: 'Mystery' },
  { id: 10749, name: 'Romance' },
  { id: 878, name: 'Science Fiction' },
  { id: 10770, name: 'TV Movie' },
  { id: 53, name: 'Thriller' },
  { id: 10752, name: 'War' },
  { id: 37, name: 'Western' },
];
const getMainMovieDetails = jest.fn(() => ({}));
test('MovieItem renders correctly without poster image', () => {
  const curr = {
    adult: false,
    backdrop_path: null,
    genre_ids: [12, 878, 28],
    id: 299534,
    original_language: 'en',
    original_title: 'Avengers: Endgame',
    overview:
      "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",
    popularity: 323.106,
    poster_path: null,
    release_date: '2019-04-24',
    title: 'Avengers: Endgame',
    video: false,
    vote_average: 8.6,
    vote_count: 4484,
  };
  const movies = {};

  const renderer = new ShallowRenderer();
  renderer.render(
    <MovieItem
      curr={curr}
      getMainMovieDetails={getMainMovieDetails}
      genres={genres}
      movies={movies}
    />,
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('MovieItem renders correctly with poster image', () => {
  const curr = {
    genre_ids: [12, 878, 28],
    original_title: 'Avengers: Endgame',
    overview:
      "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",
    poster_path: 'testImagePath',
    title: 'Avengers: Endgame',
  };
  const movies = {};
  const renderer = new ShallowRenderer();
  const result = renderer.render(
    <MovieItem
      curr={curr}
      genres={genres}
      getMainMovieDetails={getMainMovieDetails}
      movies={movies}
      typeView="cards"
    />,
  );
  expect(result).toMatchSnapshot();
});

test('MovieItem renders correctly with lines classes', () => {
  const curr = {
    genre_ids: [12, 878, 28],
    original_title: 'Avengers: Endgame',
    overview:
      "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",

    poster_path: 'testImagePath',

    title: 'Avengers: Endgame',
  };
  const movies = {};

  const renderer = new ShallowRenderer();
  const result = renderer.render(
    <MovieItem
      curr={curr}
      genres={genres}
      movies={movies}
      getMainMovieDetails={getMainMovieDetails}
      typeView="lines"
    />,
  );
  expect(result).toMatchSnapshot();
});

describe('MovieItem: selectGenre', () => {
  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  it('MovieItem: kewDown on genre, expect props', async () => {
    const idGenre = 35;
    fetchMock
      .getOnce(
        `https://api.themoviedb.org/3/discover/movie?api_key=75331f1a740385460b25b56203149aa8&with_genres=${idGenre}`,
        {
          headers: { 'content-type': 'application/json' },
          body: { page: 1, results: [1, 2, 3], status: 'ok' },
        },
      )
      .catch(err => err);
    const props = {
      genres: [{ id: 31, name: 'Action' }, { id: 35, name: 'Drama' }],
      fetchMoviesOnGenre: requestsFilms.fetchMoviesOnGenre,
      history: { push: jest.fn() },
    };
    const e = {
      key: 'a',
      target: {
        id: 'test',
        textContent: 'Drama',
      },
    };
    const result = keydonwGenres(props, e);
    expect(result).toEqual(props);
  });

  it('MovieItem: kewDown Enter, expect data', async () => {
    const idGenre = 35;
    fetchMock
      .getOnce(
        `https://api.themoviedb.org/3/discover/movie?api_key=75331f1a740385460b25b56203149aa8&with_genres=${idGenre}`,
        {
          headers: { 'content-type': 'application/json' },
          body: { page: 1, results: [1, 2, 3], status: 'ok' },
        },
      )
      .catch(err => err);
    const props = {
      genres: [{ id: 31, name: 'Action' }, { id: 35, name: 'Drama' }],
      fetchListMovies: requestsFilms.fetchListMovies,
      history: { push: jest.fn() },
    };
    const e = {
      key: 'Enter',
      target: {
        id: 'test',
        textContent: 'Drama',
      },
    };
    const result = keydonwGenres(props, e);
    expect(result).toEqual(props);
  });

  it('MovieItem: kewDown on genre', async () => {
    const idGenre = 35;
    fetchMock
      .getOnce(
        `https://api.themoviedb.org/3/discover/movie?api_key=75331f1a740385460b25b56203149aa8&with_genres=${idGenre}`,
        {
          headers: { 'content-type': 'application/json' },
          body: { page: 1, results: [1, 2, 3], status: 'ok' },
        },
      )
      .catch(err => err);
    const props = {
      genres: [{ id: 31, name: 'Action' }, { id: 35, name: 'Drama' }],
      fetchMoviesOnGenre: requestsFilms.fetchMoviesOnGenre,
    };
    const e = {
      key: 'a',
      target: {
        id: 'test',
        textContent: 'Drama',
      },
    };
    const result = keydonwGenres(props, e);
    expect(result).toEqual(props);
  });
});
describe('changeMainMovie - getMainMovieDetails would be called', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    container.id = 'modalRoot';
    document.body.appendChild(container);
  });
  // it('getMainMovieDetails - test', () => {
  //   const props = {
  //     genres,
  //     getMainMovieDetails,
  //     curr: {
  //       backdrop_path: null,
  //       genre_ids: [12, 878, 28],
  //       id: 299534,
  //       original_title: 'Avengers: Endgame',
  //       overview:
  //         "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",
  //       poster_path: null,
  //       title: 'Avengers: Endgame',
  //     },
  //   };
  //   act(() => {
  //     ReactDOM.render(
  //       <Router>
  //         <MovieItem {...props} />
  //       </Router>,
  //       container,
  //     );
  //   });

  //   jest.spyOn(props, 'getMainMovieDetails');
  //   const node1 = container.querySelector('figcaption button');
  //   ReactTestUtils.Simulate.click(node1);
  //   expect(getMainMovieDetails).toHaveBeenCalled();
  // });
});

describe('shouldComponentUpdate', () => {
  describe('shouldComponentUpdate', () => {
    const initial = { fullOverwie: true };
    const initial2 = { fullOverwie: false };

    it('MovieDetailsPage: check shouldComponentUpdate to BE TRUE', () => {
      jest.spyOn(MovieItem.prototype, 'shouldComponentUpdate');
      const action = MovieItem.prototype.shouldComponentUpdate.call(
        { state: { ...initial } },
        null,
        { ...initial2 },
      );
      expect(action).toBeTruthy();
    });

    it('MovieDetailsPage: check shouldComponentUpdate to BE FALSE', () => {
      jest.spyOn(MovieItem.prototype, 'shouldComponentUpdate');
      const action = MovieItem.prototype.shouldComponentUpdate.call(
        { state: { ...initial } },
        null,
        { ...initial },
      );
      expect(action).toBeFalsy();
    });
  });
});

describe('events', () => {
  const container = document.createElement('div');
  container.id = 'wrapper';
  const props = {
    curr: {
      adult: false,
      backdrop_path: null,
      genre_ids: [12, 878, 28],
      id: 299534,
      original_language: 'en',
      original_title: 'Avengers: Endgame',
      overview:
        "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",
      popularity: 323.106,
      poster_path: null,
      release_date: '2019-04-24',
      title: 'Avengers: Endgame',
      video: false,
      vote_average: 8.6,
      vote_count: 4484,
    },
    movies: {},
    genres,
    getMainMovieDetails,
  };
  beforeEach(() => {
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(
        <Router>
          <MovieItem {...props} />
        </Router>,
        container,
      );
    });
  });

  it('hoverOverwieBlock', () => {
    const node = document.querySelector('#showOverwie');
    ReactTestUtils.Simulate.click(node);
    const result = document.querySelector('#wrapper');
    expect(result).toMatchSnapshot();
  });

  it('hoverOverwieBlock', () => {
    const node = document.querySelector('#showOverwie');
    ReactTestUtils.Simulate.click(node);
    const node2 = document.querySelector('#closeHover');
    ReactTestUtils.Simulate.click(node2);
    const result = document.querySelector('#wrapper');
    expect(result).toMatchSnapshot();
  });
});
