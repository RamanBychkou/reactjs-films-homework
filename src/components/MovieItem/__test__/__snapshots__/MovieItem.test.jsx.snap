// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem renders correctly with lines classes 1`] = `
<article
  className="movieContainer lines"
  onMouseLeave={[Function]}
>
  <figure
    className="figcaptionBlock"
  >
    <div>
      <img
        alt="Avengers: Endgame"
        src="https://image.tmdb.org/t/p/w1280undefined"
      />
    </div>
    <figcaption>
      <div
        className="movieTitle"
      >
        <button
          href="#"
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          <Link
            to={
              Object {
                "pathname": "/films/undefined",
              }
            }
          >
            Avengers: Endgame
          </Link>
        </button>
        <span />
      </div>
      <div
        className="movieGenres"
      >
        <ul>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={12}
            >
              Adventure,
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={878}
            >
              Science
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={28}
            >
              Fiction,
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              Action
            </button>
          </li>
        </ul>
      </div>
    </figcaption>
  </figure>
  <hoverMovieCard
    fetchVideo={[Function]}
    fullOverwie={false}
    genres={
      Array [
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={12}
          >
            Adventure,
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={878}
          >
            Science
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={28}
          >
            Fiction,
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Action
          </button>
        </li>,
      ]
    }
    hoverOverwieBlock={[Function]}
    hoverOverwieNone={[Function]}
    id={0}
    movies={Object {}}
    overview="After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store."
    rate={0}
    title="Avengers: Endgame"
  />
</article>
`;

exports[`MovieItem renders correctly with poster image 1`] = `
<article
  className="movieContainer cards"
  onMouseLeave={[Function]}
>
  <figure
    className="figcaptionBlock"
  >
    <div>
      <img
        alt="Avengers: Endgame"
        src="https://image.tmdb.org/t/p/w300testImagePath"
      />
    </div>
    <figcaption>
      <div
        className="movieTitle"
      >
        <button
          href="#"
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          <Link
            to={
              Object {
                "pathname": "/films/undefined",
              }
            }
          >
            Avengers: Endgame
          </Link>
        </button>
        <span />
      </div>
      <div
        className="movieGenres"
      >
        <ul>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={12}
            >
              Adventure,
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={878}
            >
              Science
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={28}
            >
              Fiction,
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              Action
            </button>
          </li>
        </ul>
      </div>
    </figcaption>
  </figure>
  <hoverMovieCard
    fetchVideo={[Function]}
    fullOverwie={false}
    genres={
      Array [
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={12}
          >
            Adventure,
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={878}
          >
            Science
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={28}
          >
            Fiction,
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Action
          </button>
        </li>,
      ]
    }
    hoverOverwieBlock={[Function]}
    hoverOverwieNone={[Function]}
    id={0}
    movies={Object {}}
    overview="After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store."
    rate={0}
    title="Avengers: Endgame"
  />
</article>
`;

exports[`MovieItem renders correctly without poster image 1`] = `
<article
  className="movieContainer undefined"
  onMouseLeave={[Function]}
>
  <figure
    className="figcaptionBlock"
  >
    <div>
      <img
        alt="Avengers: Endgame"
        src="https://api.ballotpedia.org/v3/thumbnail/"
      />
    </div>
    <figcaption>
      <div
        className="movieTitle"
      >
        <button
          href="#"
          onKeyDown={[Function]}
          tabIndex="0"
          type="button"
        >
          <Link
            id={299534}
            to={
              Object {
                "pathname": "/films/299534",
              }
            }
          >
            Avengers: Endgame
          </Link>
        </button>
        <span>
          8.6
        </span>
      </div>
      <div
        className="movieGenres"
      >
        <ul>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={12}
            >
              Adventure,
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={878}
            >
              Science
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
              value={28}
            >
              Fiction,
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              Action
            </button>
          </li>
        </ul>
      </div>
    </figcaption>
  </figure>
  <hoverMovieCard
    fetchVideo={[Function]}
    fullOverwie={false}
    genres={
      Array [
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={12}
          >
            Adventure,
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={878}
          >
            Science
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
            value={28}
          >
            Fiction,
          </button>
        </li>,
        <li>
          <button
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            Action
          </button>
        </li>,
      ]
    }
    hoverOverwieBlock={[Function]}
    hoverOverwieNone={[Function]}
    id={299534}
    movies={Object {}}
    overview="After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store."
    rate={8.6}
    title="Avengers: Endgame"
  />
</article>
`;

exports[`events hoverOverwieBlock 1`] = `
<div
  id="wrapper"
>
  <article
    class="movieContainer undefined"
  >
    <figure
      class="figcaptionNone"
    >
      <div>
        <img
          alt="Avengers: Endgame"
          src="https://api.ballotpedia.org/v3/thumbnail/"
        />
      </div>
      <figcaption>
        <div
          class="movieTitle"
        >
          <button
            href="#"
            tabindex="0"
            type="button"
          >
            <a
              href="/films/299534"
              id="299534"
            >
              Avengers: Endgame
            </a>
          </button>
          <span>
            8.6
          </span>
        </div>
        <div
          class="movieGenres"
        >
          <ul>
            <li>
              <button
                type="button"
                value="12"
              >
                Adventure,
              </button>
            </li>
            <li>
              <button
                type="button"
                value="878"
              >
                Science
              </button>
            </li>
            <li>
              <button
                type="button"
                value="28"
              >
                Fiction,
              </button>
            </li>
            <li>
              <button
                type="button"
              >
                Action
              </button>
            </li>
          </ul>
        </div>
      </figcaption>
    </figure>
    <div
      class="hoverMovie"
    >
      <div
        class="hoverControlNone"
      >
        <label
          for="playTrailer"
          id="299534"
        >
          <input
            class="playTrailer"
            id="299534"
            name="playTrailer"
            type="button"
          />
          Watch Now
        </label>
        <button
          class="showOverwie"
          id="showOverwie"
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        class="hoverOverwie hoverOverwieBlock"
      >
        <button
          class="hoverClose"
          id="closeHover"
          type="button"
        >
          ╳
        </button>
        <div
          class="hoverOverwieTitle"
        >
          <div
            class="hoverOverwieTitleItem"
          >
            <h3>
              Avengers: Endgame
            </h3>
            <ul>
              <li>
                <button
                  type="button"
                  value="12"
                >
                  Adventure,
                </button>
              </li>
              <li>
                <button
                  type="button"
                  value="878"
                >
                  Science
                </button>
              </li>
              <li>
                <button
                  type="button"
                  value="28"
                >
                  Fiction,
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Action
                </button>
              </li>
            </ul>
          </div>
          <div
            class="hoverOverwieTitleItem"
          >
            <p>
              8.6
            </p>
          </div>
        </div>
        <p>
          After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.
        </p>
        <button
          id="299534"
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
  </article>
</div>
`;

exports[`events hoverOverwieBlock 2`] = `
<div
  id="wrapper"
>
  <article
    class="movieContainer undefined"
  >
    <figure
      class="figcaptionBlock"
    >
      <div>
        <img
          alt="Avengers: Endgame"
          src="https://api.ballotpedia.org/v3/thumbnail/"
        />
      </div>
      <figcaption>
        <div
          class="movieTitle"
        >
          <button
            href="#"
            tabindex="0"
            type="button"
          >
            <a
              href="/films/299534"
              id="299534"
            >
              Avengers: Endgame
            </a>
          </button>
          <span>
            8.6
          </span>
        </div>
        <div
          class="movieGenres"
        >
          <ul>
            <li>
              <button
                type="button"
                value="12"
              >
                Adventure,
              </button>
            </li>
            <li>
              <button
                type="button"
                value="878"
              >
                Science
              </button>
            </li>
            <li>
              <button
                type="button"
                value="28"
              >
                Fiction,
              </button>
            </li>
            <li>
              <button
                type="button"
              >
                Action
              </button>
            </li>
          </ul>
        </div>
      </figcaption>
    </figure>
    <div
      class="hoverMovie"
    >
      <div
        class="hoverControlBlock"
      >
        <label
          for="playTrailer"
          id="299534"
        >
          <input
            class="playTrailer"
            id="299534"
            name="playTrailer"
            type="button"
          />
          Watch Now
        </label>
        <button
          class="showOverwie"
          id="showOverwie"
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        class="hoverOverwie hoverOverwieNone"
      >
        <button
          class="hoverClose"
          id="closeHover"
          type="button"
        >
          ╳
        </button>
        <div
          class="hoverOverwieTitle"
        >
          <div
            class="hoverOverwieTitleItem"
          >
            <h3>
              Avengers: Endgame
            </h3>
            <ul>
              <li>
                <button
                  type="button"
                  value="12"
                >
                  Adventure,
                </button>
              </li>
              <li>
                <button
                  type="button"
                  value="878"
                >
                  Science
                </button>
              </li>
              <li>
                <button
                  type="button"
                  value="28"
                >
                  Fiction,
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Action
                </button>
              </li>
            </ul>
          </div>
          <div
            class="hoverOverwieTitleItem"
          >
            <p>
              8.6
            </p>
          </div>
        </div>
        <p>
          After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.
        </p>
        <button
          id="299534"
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
  </article>
</div>
`;
